plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -64,10 +65,15 @@ func (c *X509Cert) SampleConfig() string {
 }
 
 func (c *X509Cert) sourcesToURLs() error {
+	uriregexp, err := regexp.Compile("^(((udp|tcp)[46]?)|https)://")
+	if err != nil {
+		return fmt.Errorf("could not compile uri regexp %v", err)
+	}
+
 	for _, source := range c.Sources {
-		if strings.HasPrefix(source, "file://") ||
+		if !uriregexp.MatchString(source) && (strings.HasPrefix(source, "file://") ||
```

Comment:
Maybe I miss something, but shouldn't `uriregexp.MatchString(source)` always be `false` if any of the other conditions is true? I think it is redundant, isn't it?

Reply:
Original code is: https://github.com/influxdata/telegraf/blob/master/plugins/inputs/x509_cert/x509_cert.go#L68-L70
and AFAIK `strings.Index(source, ":\\") != 1` will match pretty much everything (and current master complains for https:// and tcp:// sources):
```
[inputs.x509_cert] could not find file: &{https://example.org:443 false false }
[inputs.x509_cert] could not find file: &{tcp://2.example.org:443 false false }
```
I'm not quite sure what the purpose of the strings.Index(source, ":\\") != 1 is, but 
maybe this could be rewritten without regexp (untested):
```
if (strings.HasPrefix(source, "file://") ||
   strings.HasPrefix(source, "/")) &&
   !(strings.Index(source, ":\\") == 1) {
```

Reply:
OMG! I think `strings.Index(source, ":\\") != 1` this is just the inverse of what it is supposed to be. Making this `strings.Index(source, ":\\") == 1` should be correct to cover `C:\\` type of paths.

Reply:
Then
```
if strings.Index(source, ":\\") == 1 {
    source = "file://" + filepath.ToSlash(source)
}
```
(https://github.com/influxdata/telegraf/blob/master/plugins/inputs/x509_cert/x509_cert.go#L78-L80)
is redundant and will never match ?

Do you know what the intention of the original code is ? Should it try to find fileglobs for sources like `C:\something` ? Now I think it tries to find globs for sources starting with `/` or `file://` and treat `C:\something` as `file://C:\something` (without trying to glob) ?

Reply:
Ok I walked through the PR adding this and it seems the discussion point was if Windows actually is supported by `globpath`. If it isn't, the code should be
```golang
		if strings.HasPrefix(source, "file://") ||
		   strings.HasPrefix(source, "/") {
			source = filepath.ToSlash(strings.TrimPrefix(source, "file://"))
			g, err := globpath.Compile(source)
			if err != nil {
				return fmt.Errorf("could not compile glob %v: %v", source, err)
			}
			c.globpaths = append(c.globpaths, g)
		} else {
			if strings.Index(source, ":\\") == 1 {
				source = "file://" + filepath.ToSlash(source)
			}
			u, err := url.Parse(source)
			if err != nil {
				return fmt.Errorf("failed to parse cert location - %s", err.Error())
			}
			c.locations = append(c.locations, u)
		}
```
Otherwise it should be 
```golang
		if strings.HasPrefix(source, "file://") ||
		   strings.HasPrefix(source, "/") || 
		   strings.Index(source, ":\\") == 1 {
			source = filepath.ToSlash(strings.TrimPrefix(source, "file://"))
			g, err := globpath.Compile(source)
			if err != nil {
				return fmt.Errorf("could not compile glob %v: %v", source, err)
			}
			c.globpaths = append(c.globpaths, g)
		} else {
			u, err := url.Parse(source)
			if err != nil {
				return fmt.Errorf("failed to parse cert location - %s", err.Error())
			}
			c.locations = append(c.locations, u)
		}
```

Reply:
I updated the code with expectation that `globpath` works on windows. (I'll try to test this with the windows binaries from CI build and if I can't get globpath to work with something like `C:\temp\*.pem` I'll change it to version that ignores globpath for `driveletter:\`).